### ISSUE TITLE
How to limit the number of tasks in a session? Have this function?

### ISSUE THREAD

**Issue by zbooa** [arvidn/libtorrent#4578]
Add torrent to session, add magnet, 。。。。。
Is there a function to limit the number of simultaneous tasks?
Like 
Simultaneous tasks are limited to 5, and tasks that exceed the number limit require other tasks to complete before they can start

**Comment by arvidn** [comment]
what do you mean by "task"?

**Comment by zbooa** [comment]
mean lt::torrent_handle.

**Comment by arvidn** [comment]
sounds like you're looking for queing: https://libtorrent.org/manual-ref.html#queuing

**Comment by zbooa** [comment]
> sounds like you're looking for queing: https://libtorrent.org/manual-ref.html#queuing

Yes, that's exactly what I want. thank you. But I tried and it didn't work.

```
pack.set_int(lt::settings_pack::active_limit, 1);
pack.set_int(lt::settings_pack::auto_manage_interval, 3);

pack.set_int(lt::settings_pack::active_checking, 1);
pack.set_int(lt::settings_pack::active_downloads, 1);
pack.set_int(lt::settings_pack::active_seeds, 1);
```

There is no limit to the number of downloaded torrents. Trying to added it when add torrent:
`p.flags &= lt::torrent_flags::auto_managed;`
The torrent cant download... how setting this queuing? have examples?

**Comment by arvidn** [comment]
I don't think you should mask the flags in `add_torrent_params` to *just* `auto_managed`. Instead, make sure that flag is *set*.

`p.flags |= torrent_flags::auto_managed;`

If you still have multiple started torrent, make sure you don't have the `dont_count_slow_torrents` setting enabled.

**Comment by zbooa** [comment]
> I don't think you should mask the flags in `add_torrent_params` to _just_ `auto_managed`. Instead, make sure that flag is _set_.
> 
> `p.flags |= torrent_flags::auto_managed;`
> 
> If you still have multiple started torrent, make sure you don't have the `dont_count_slow_torrents` setting enabled.

What is the status of waiting to download in the queue?
`lt::torrent_status::queued_for_checking` ？？

**Comment by arvidn** [comment]
`lt::torrent_status::downloading`

**Comment by zbooa** [comment]
> `lt::torrent_status::downloading`

If so, in the queue, downloading and waiting for downloading are the same state, how can I distinguish them?

**Comment by arvidn** [comment]
one has the `paused` flag and one doesn't. https://libtorrent.org/reference-Core.html#torrent_flags_t

**Comment by zbooa** [comment]
Thank you. 
I understand that `auto_managed` cannot be suspended manually. It is started and managed by libtorrent.
But I still don't understand that in the queue, all torrents are `auto_managed`. Are the torrents being downloaded has `pause` or are the torrents waiting has `pause`? How code it?

**Comment by arvidn** [comment]
paused mean the torrent is not active, not downloading or uploading anything. I don't know what you mean by "auto_managed cannot be suspended manually". You can change torrent flags on a torrent.

**Comment by zbooa** [comment]
> paused mean the torrent is not active, not downloading or uploading anything. I don't know what you mean by "auto_managed cannot be suspended manually". You can change torrent flags on a torrent.

Let me try again, thank you

**Comment by zbooa** [comment]
> paused mean the torrent is not active, not downloading or uploading anything. I don't know what you mean by "auto_managed cannot be suspended manually". You can change torrent flags on a torrent.

So, in the queue, Torrent waiting to download, status is `lt::torrent_status::downloading`,  and it doesn't has the `paused` flag ? Is it?

**Comment by arvidn** [comment]
"paused mean the torrent is not active, not downloading or uploading anything."

**Comment by zbooa** [comment]
> "paused mean the torrent is not active, not downloading or uploading anything."

OK.
